python/console/console_editor.py

```diff
@@ -406,7 +406,7 @@ def save(self, filename=None):
         if self.isReadOnly():
             return
         tabwidget = self.tabwidget
-        index = tabwidget.indexOf(self)
+        index = tabwidget.indexOf(self.parent)
```

Comment:
is this an accident too?

Reply:
Yes it is, and that is actually why the name of the tab was not updated :). I had too many branches on the python console at some point, and I made kind of a mess switching among them, sorry.